app/components/avo/views/resource_index_component.rb

```diff
@@ -135,8 +135,12 @@ def show_search_input
 
   def authorized_to_search?
     # Hide the search if the authorization prevents it
-    return true unless resource.authorization.respond_to?(:has_action_method?)
-    return false unless resource.authorization.has_action_method?("search")
+    if @resource.class.authorization_policy.present?
+      return true unless resource.authorization.respond_to?(:has_action_method?)
+      return false unless resource.authorization.has_action_method?("search")
+    else
+      return true
+    end
```

Comment:
```suggestion
    return true if @resource.class.authorization_policy.blank?
    return true unless resource.authorization.respond_to?(:has_action_method?)
    return false unless resource.authorization.has_action_method?("search")
```

Let's favor the use of guard clauses over  "if-else" statements. Guard clauses enhance code readability and maintainability.

Reply:
Thanks. I will update it.